appinventor/components/src/com/google/appinventor/components/runtime/ImagePicker.java

```diff
@@ -179,12 +180,14 @@ private void copyToExternalStorageAndDeleteSource(File source, String extension)
 
     File dest = null;
 
-    String fullDirname = QUtil.getExternalStoragePath(container.$form()) + imagePickerDirectoryName;
-    File destDirectory = new File(fullDirname);
+    String fullDirname = container.$form().getDefaultPath(imagePickerDirectoryName);
```

Comment:
I take it that is due to the changes over time in the existence and APIs related to external storage and permissions related to file access?

Reply:
More specifically, we introduced DefaultFileScope as a way for users to adjust where files should end up by default when created by components, but QUtil.getExternalStoragePath doesn't take this into account and basically forces files to be put into the external storage even if the developer opts for private storage. By making this change, we account for the scope selection by the user, but it also causes minor complications because getExternalStoragePath returns a UNIX-like path but getDefaultPath returns a file: URI. That's my fault because I added both (about a year apart) and didn't think to make the behavior consistent in terms of the return value.

Reply:
Got it!  Thanks.